src/oph/heratepalvelu/tep/rahoitusLaskentaHandler.clj

```diff
@@ -256,14 +184,14 @@
                                (:opiskeluoikeus-oid herate))]
           (if (some? opiskeluoikeus)
             (let [koulutustoimija (c/get-koulutustoimija-oid opiskeluoikeus)]
-              (if (some? (tep-herate-checker herate))
-                (log/error {:herate herate :msg (tep-herate-checker herate)})
-                (when (and (c/check-opiskeluoikeus-tila opiskeluoikeus
-                                                        (:loppupvm herate))
-                           (check-not-fully-keskeytynyt herate)
-                           (c/check-opiskeluoikeus-suoritus-types?
-                             opiskeluoikeus)
-                           (c/check-sisaltyy-opiskeluoikeuteen? opiskeluoikeus))
+              (if (some? (jh/tep-herate-checker herate))
+                (log/error {:herate herate :msg (jh/tep-herate-checker herate)})
+                (when-not
+                 (or (c/terminaalitilassa? opiskeluoikeus (:loppupvm herate))
+                     (jh/fully-keskeytynyt? herate)
+                     (not (c/has-one-or-more-ammatillinen-tutkinto?
```

Comment:
täällä sama juttu.  Sen sijaan (when) voitaisiin kokonaan tiputtaa pois, koska (and) ajaa saman asian.